tests/config/test_find_default_config_files.py

```diff
@@ -223,3 +224,17 @@ def test_cfg_has_config(content: str, expected: str, tmp_path: Path) -> None:
     with open(fake_cfg, "w", encoding="utf8") as f:
         f.write(content)
     assert _cfg_has_config(fake_cfg) == expected
+
+
+def test_non_existent_home() -> None:
+    """Test that we handle a non-existent home directory.
+
+    Reported in https://github.com/PyCQA/pylint/issues/6802.
+    """
+    with mock.patch("pathlib.Path.home", side_effect=RuntimeError):
```

Comment:
`Path.home()` has been re-implemented on every version between 3.7-3.10 😅 

So it's almost impossible to write one good test cases that fits them all. This was much easier.

All versions do raise `RuntimeError`.